stmhal/Makefile

```diff
@@ -321,6 +321,16 @@ $(BUILD)/%_$(BOARD).c $(HEADER_BUILD)/%.h $(HEADER_BUILD)/%_af_const.h $(BUILD)/
 $(BUILD)/pins_$(BOARD).o: $(BUILD)/pins_$(BOARD).c
 	$(call compile_c)
 
+GEN_STMCONST_HDR = $(BUILD)/modstm_const.gen.h
```

Comment:
> It should go in HEADER_BUILD dir (ie genhdr/) and be called modstm_const.h (without .gen, not needed).

ok - It wasn't a standalone header, so that's why I was leaning to having it out.

We should probably move pins_qstr.h into genhdr as well to be consistent (I'll do that as a separate PR)